### PR TITLE
add editing for 'article' type posts

### LINKED ISSUE #1549: Article: edit article
As a user, I'd like to be able edit an article after I've created it, similar to how I can edit a post.

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
closes #1549

#### What's this PR do?

this adds the ability to edit an 'article' type post.

#### How should this be manually tested?

you should be able to edit an article post. to be able to edit articles, you can follow the setup instructions on the PR that added it, here: https://github.com/mitodl/open-discussions/pull/1557

then you should be able to navigate to an article post and edit it. you should also be able to edit existing text posts as normal, and you should be able to make comments and reply to comments.

### CODE REVIEW

File: static/js/components/CommentForms_test.js [hunk 1]
```diff
@@ -612,11 +614,17 @@ describe("CommentForms", () => {
       wrapper = renderEditPostForm()
     })
 
-    it("should use the <Editor /> rather than a textarea", () => {
+    it("should use the <Editor /> rather than a textarea if text", () => {
```
Comment (mbertrand) [inline static/js/components/CommentForms_test.js]: `rather than a textarea if a post` ?

Comment (aliceriot) [reply static/js/components/CommentForms_test.js]: haha good catch

**Comment by aliceriot** [comment]
yeah I meant to fix that `appendChild` error, I'll add some checks to fix that now

**Comment by aliceriot** [comment]
@mbertrand ok that should be fixed now

**Pull request merged**